api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-blm/marxan-sandbox-blm-runner.service.ts

```diff
@@ -1,66 +1,132 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 import { JobData } from '@marxan/blm-calibration';
 
 import { SandboxRunner } from '../sandbox-runner';
-import { WorkspaceBuilder } from '../ports/workspace-builder';
-import { SandboxRunnerOutputHandler } from '../sandbox-runner-output-handler';
 import { BlmInputFiles } from './blm-input-files';
+import { MarxanRun } from '@marxan-geoprocessing/marxan-sandboxed-runner/marxan-run';
+import { Cancellable } from '@marxan-geoprocessing/marxan-sandboxed-runner/ports/cancellable';
+import AbortController from 'abort-controller';
+import { BlmResultsParser } from '@marxan-geoprocessing/marxan-sandboxed-runner/adapters-blm/blm-results.parser';
 
 @Injectable()
 export class MarxanSandboxBlmRunnerService
   implements SandboxRunner<JobData, void> {
+  readonly #controllers: Record<string, AbortController> = {};
+  readonly #logger = new Logger(this.constructor.name);
   constructor(
-    private readonly workspaceService: WorkspaceBuilder,
     private readonly inputFilesHandler: BlmInputFiles,
-    private readonly outputFilesHandler: SandboxRunnerOutputHandler<void>,
+    private readonly blmResultsParser: BlmResultsParser,
```

Comment:
`MarxanSandboxBlmRunnerService` now depends on concrete implementation instead abstraction. Please make it use `SandboxRunnerOutputHandler` again and provide concrete implementation within adapters.